### PR TITLE
Implement unary expression for vector types.

### LINKED ISSUE #956: Allow unary expressions for vectors
Currently unary expressions are only allowed for scalars.

### PR DESCRIPTION
**Pull request by fabxc**
Closes #956

@juliusv @brian-brazil

### CODE REVIEW

File: promql/parse_test.go [hunk 1]
```diff
@@ -192,13 +212,17 @@ var testExpr = []struct {
...
 	}, {
-		input:  `-"string"`,
+		input:  `-test[5m]`,
 		fail:   true,
-		errMsg: "expected type scalar in unary expression, got string",
+		errMsg: `unary expression only allowed on expressions of type scalar or vector, got "matrix"`,
+	}, {
+		input:  `*test`,
+		fail:   true,
+		errMsg: "no valid expression found",
```
Comment (brian-brazil) [inline promql/parse_test.go]: Shouldn't `-If` on line 989 no longer fail now?

File: promql/parse.go [hunk 2]
```diff
@@ -1075,7 +1075,10 @@ func (p *parser) checkType(node Node) (typ ExprType) {
 		if n.Op != itemADD && n.Op != itemSUB {
 			p.errorf("only + and - operators allowed for unary expressions")
 		}
-		p.expectType(n.Expr, ExprScalar, "unary expression")
+		if t := p.checkType(n.Expr); t != ExprScalar && t != ExprVector {
+			p.errorf("unary expression only allowed on expressions of type scalar or vector, got %q", t)
+		}
+		// p.expectType(n.Expr, ExprScalar, "unary expression")
```
Comment (juliusv) [inline promql/parse.go]: Remove commented-out line?

File: promql/parse_test.go (hunk 1 above)
Comment (fabxc) [inline promql/parse_test.go]: If is a keyword. Keywords are not valid metric names so this fails for another reason - still a regression test from the fuzzing.

Comment (brian-brazil) [inline promql/parse_test.go]: https://github.com/prometheus/prometheus/issues/963 filed to cover this

**Comment by brian-brazil** [comment]
:+1:

**Pull request merged**